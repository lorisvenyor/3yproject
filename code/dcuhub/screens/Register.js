import React from 'react'
import { StyleSheet, Text, View, Image, Dimensions, TextInput, Button, TouchableOpacity,Alert} from 'react-native'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { StatusBar } from 'expo-status-bar'
import * as firebase from 'firebase'
import {SafeAreaView} from 'react-native-safe-area-context'
import Button2 from '../shared/button_2'

const DeviceHeight = Dimensions.get('screen').height;

export default class Register extends React.Component {

    state = {
        name:"",
        email: "",
        password: "",
        errorMessage: null
    };
 
    handleSignUp = async () => {
        const { email, password, name } = this.state;
        const { navigate } = this.props.navigation

        // exception handling, checks if one of the inputs is empty
        if(email == '' || name == '' || password == ''){
            Alert.alert('Empty States','Please complete the form.')
        }
        else if(email.split('@')[1] != 'mail.dcu.ie'){
            Alert.alert("Invalid Email","This domain is not recognized. Please enter your DCU email.")
        }else{
        firebase.auth().createUserWithEmailAndPassword(this.state.email, this.state.password)
        .then(userCredentials => {
            userCredentials.user.sendEmailVerification()
            Alert.alert("Verification Email", "verification email has been sent to your email.")
            firebase.auth().signOut()

            return userCredentials.user.updateProfile({
                displayName: this.state.name
            });
        })
        .catch(error => this.setState({errorMessage: error.message}))
    }
    
}
    render () {
        return (
            <SafeAreaView style={styles.container}>
            <StatusBar style="auto"/>
            <KeyboardAwareScrollView>

                    <View style={styles.image_view}>  
                        <Image 
                        source={require("../assets/images/dcuhublogo_higherres.png")}
                        resizeMode="contain"
                        resizeMode="center"
                        style={styles.image_style}>
                        </Image>
                    </View> 

                    <View style={{
                        alignSelf: 'center',
                        marginTop: -50,
                        marginBottom: 50
                    }}>
                        <Text style={{
                            textAlign: 'center',
                            fontSize: 22,
                            color: 'black'
                        }}>Welcome to DCU Hub!</Text>
                    </View>

                    <View style={{
                        alignSelf: 'center',
                        marginTop: -48,
                        marginBottom: 50
                    }}>
                        <Text style={{
                            textAlign: 'center',
                            fontSize: 17,
                            color: '#696868'
                        }}>Sign up to get started.</Text>
                    </View>

                    {/* errors from firebase popping up on screen */}
                    <View style={styles.errorMessage}>
                        {this.state.errorMessage && <Text style={styles.error}>{this.state.errorMessage}</Text>}
                    </View>

                   < View style={styles.top_placeholder}>
                        <TextInput 
                            style={styles.placeholder_text}
                            placeholder="FULL NAME" autoCapitalize="none" onChangeText={name => this.setState({name})}
                            value={this.state.name}>
                        </TextInput>
                    </View>

                    <View style={styles.placeholder_style}>
                        <TextInput 
                            style={styles.placeholder_text}
                            placeholder="DCU EMAIL" autoCapitalize="none" onChangeText={email => this.setState({email})}
                            value={this.state.email}
                            ></TextInput>
                    </View>

                                        
                    <View style={styles.placeholder_style}>
                        <TextInput 
                            style={styles.placeholder_text}
                            secureTextEntry={true}  // SECURE PASSWORD ie hides the texts
                            placeholder="PASSWORD" secureTextEntry autoCapitalize="none" onChangeText={password => this.setState({password})}
                            value={this.state.password}
                            ></TextInput>
                    </View>

                    {/* THIS ALSO USES FIREBASE AUTHENTICATION
                        ONCED SIGNED UP, IT SHOULD TAKE THE USER TO THE
                        CREATE PROFILE SECTION/SCREEN */}
                    <View>
                        <Button2 text='SIGN UP' onPress={this.handleSignUp}/>
                    </View>


                    <View style={styles.to_signin}>
                        <Text style={{fontSize: 18}}> Already have an account? </Text>
                        <TouchableOpacity onPress={() => this.props.navigation.navigate('Login')}>
                            <Text style={{color: "#d40059", fontSize: 18}}> Sign In </Text>
                        </TouchableOpacity>
                    </View>
            
            </KeyboardAwareScrollView>
            </SafeAreaView>
        );
    }
}


const styles = StyleSheet.create({
    container : {
        flex: 1,
        height: DeviceHeight,
        backgroundColor: "white"
    },
    placeholder_style: {
        marginHorizontal: 40,
        borderWidth: 1.5,
        marginTop:15,
        paddingHorizontal:10,
        borderColor:"#25afcd",
        borderRadius:10,
        paddingVertical: 10// size of the placeholder
    }, 
    top_placeholder: {
        marginHorizontal: 40,
        borderWidth: 1.5,
        marginTop: -10,
        paddingHorizontal:10,
        borderColor:"#25afcd",
        borderRadius: 10,
        paddingVertical: 10
    },
    placeholder_text: {
        paddingHorizontal:10
    }, 
    errorMessage: {
        alignItems: "center",
        justifyContent: "center",
        marginHorizontal: 30,
        marginTop: -20,
        marginBottom: 30
    },
    error:{
        color: "#d40059",
        fontSize: 15,
        fontWeight: "600",
        textAlign: "center"
    },
    image_style: {
        width: Dimensions.get('screen').width, 
        height: Dimensions.get('screen').height / 2
    },
    image_view: {
        alignSelf: 'center',
        justifyContent: 'center',
        marginTop: -40
    },
    to_signin: {
        flexDirection:"row",
        alignItems:"center",
        justifyContent: "center",
        marginTop: 10,
        marginBottom: 30
    }
})
