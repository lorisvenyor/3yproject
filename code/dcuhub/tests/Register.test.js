import React from 'react'
import {create, act} from 'react-test-renderer'
import Register from '../screens/Register'


const screen = create(<Register/>)

test('Register Screen Snapshot', () => {
    expect(screen).toMatchSnapshot()
})
