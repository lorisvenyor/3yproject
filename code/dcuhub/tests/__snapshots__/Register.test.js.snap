// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Screen Snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "marginTop": -40,
          }
        }
      >
        <Image
          resizeMode="center"
          source={1}
          style={
            Object {
              "height": 667,
              "width": 750,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 50,
            "marginTop": -50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 22,
              "textAlign": "center",
            }
          }
        >
          Welcome to DCU Hub!
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 50,
            "marginTop": -48,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#696868",
              "fontSize": 17,
              "textAlign": "center",
            }
          }
        >
          Sign up to get started.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 30,
            "marginHorizontal": 30,
            "marginTop": -20,
          }
        }
      />
      <View
        style={
          Object {
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "marginHorizontal": 40,
            "marginTop": -10,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="FULL NAME"
          rejectResponderTermination={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "marginHorizontal": 40,
            "marginTop": 15,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="DCU EMAIL"
          rejectResponderTermination={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "marginHorizontal": 40,
            "marginTop": 15,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="PASSWORD"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "borderRadius": 20,
                "justifyContent": "center",
                "marginHorizontal": 100,
                "marginTop": 30,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              SIGN UP
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 30,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           Already have an account? 
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#d40059",
                "fontSize": 18,
              }
            }
          >
             Sign In 
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
